packages/pieces/community/fireflies-ai/src/lib/actions/find-meeting-by-query.ts

```diff
@@ -0,0 +1,171 @@
+import { createAction, Property } from '@activepieces/pieces-framework';
+import { AuthenticationType, httpClient, HttpMethod } from '@activepieces/pieces-common';
+import { firefliesAiAuth } from '../../index';
+import { BASE_URL } from '../common';
+
+export const findMeetingByQueryAction = createAction({
+	auth: firefliesAiAuth,
+	name: 'find_meeting_by_query',
+	displayName: 'Find Meeting by Call Deatils',
+	description: 'Searches meetings based on provided parameters.',
+	props: {
+		title: Property.ShortText({
+			displayName: 'Meeting Title',
+			required: false,
+		}),
+		hostEmail: Property.ShortText({
+			displayName: 'Host Email',
+			description: 'Filter meetings by host email.',
+			required: false,
+		}),
+		participantEmail: Property.ShortText({
+			displayName: 'Participant Email',
+			description: 'Filter meetings by participant email',
+			required: false,
+		}),
+		date: Property.DateTime({
+			displayName: 'Date',
+			description: 'Filter meetings on this date (YYYY-MM-DD).',
+			required: false,
+		}),
+	},
+	async run({ propsValue, auth }) {
+		const filterVariables: Record<string, any> = {};
+
+		if (propsValue.title) {
+			filterVariables['title'] = propsValue.title;
+		}
+
+		if (propsValue.hostEmail) {
+			filterVariables['hostEmail'] = propsValue.hostEmail;
+		}
+
+		if (propsValue.participantEmail) {
+			filterVariables['participantEmail'] = propsValue.participantEmail;
+		}
+
+		if (propsValue.date) {
+			// Convert ISO string to milliseconds for the API
+			const dateMs = new Date(propsValue.date).getTime();
+			filterVariables['date'] = dateMs;
+		}
+
+		const query = `
+			query Transcripts(
```

Comment:
queries being present directly in the actions severely impacts the readability of the code and they should have been organized better.